Cabal/doc/getting-started.rst

```diff
@@ -0,0 +1,169 @@
+
+Notice that we didn't need to run a `build` command before ``cabal run``, this
+is because ``cabal run`` first determines if the code needs to be re-built
+before running the executable. If you just want to build a target you can do so
+with ``cabal build``:
+
+``cabal build :myfirstapp``
+
+
+Adding dependencies
+^^^^^^^^^^^^^^^^^^^
+
+Next we'll add an external dependency to our application. `Hackage
+<https://hackage.haskell.org/>`__ is the Haskell community's central `package`
+archive of open source software.
+
+In our application, we'll use a package called `haskell-say
+<https://hackage.haskell.org/package/haskell-say>`__ to print text to the
+terminal with some embellishment.
+
+.. TIP::
+   If you installed ``cabal`` a while ago but haven't used it recently you may
+   need to update the package index, you can do this by running ``cabal
+   update``.
+
+In our ``myfirstapp.cabal`` file we'll update the ``build-depends`` attribute of
+the ``executable myfirstapp`` section to include ``haskell-say``:
+
+.. code-block:: cabal
+
+   executable myfirstapp
+       main-is: Main.hs
+       build-depends:
+           base >=4.11 && <4.12,
+           haskell-say ^>=1.0.0.0
+
+.. NOTE::
+   ``^>=1.0.0.0`` means use version 1.0.0.0 of the library or any more recent
+   minor release with the same major version.
+
+Next we'll update ``Main.hs`` to use the ``HaskellSay`` library:
+
+.. code-block:: haskell
+
+   module Main where
+
+   import HaskellSay (haskellSay)
+
+   main :: IO ()
+   main =
+     haskellSay "Hello, Haskell! You're using a function from another package!"
+
+``import HaskellSay (haskellSay)`` brings the ``haskellSay`` function from the
+module named ``HaskellSay`` into scope. The ``HaskellSay`` module is defined in
+the ``haskell-say`` packages that we added a dependency on above.
+
+Now you can build and re-run your code to see the new output:
+
+.. code-block:: console
+
+   $ cabal run
+       ________________________________________________________
+      /                                                        \
+     | Hello, Haskell! You're using a function from another     |
+     | package!                                                 |
+      \____       _____________________________________________/
+           \    /
+            \  /
+             \/
+       _____   _____
+       \    \  \    \
+        \    \  \    \
+         \    \  \    \
+          \    \  \    \  \-----------|
+           \    \  \    \  \          |
+            \    \  \    \  \---------|
+            /    /  /     \
+           /    /  /       \  \-------|
+          /    /  /    ^    \  \      |
+         /    /  /    / \    \  \ ----|
+        /    /  /    /   \    \
+       /____/  /____/     \____\
+
+
+What Next?
+----------
+
+Now that you know how to set up a simple Haskell package using Cabal, check out
+some of thee resources on the Haskell website's `documentation page
```

Comment:
Typo: the

Reply:
Oops, will fix in follow-up, good catch :)